src/commands/outdated.cr

```diff
@@ -3,63 +3,67 @@ require "./command"
 module Shards
   module Commands
     class Outdated < Command
+      @prereleases = false
+
       @up_to_date = true
       @output = IO::Memory.new
 
-      def self.run(path, @@prereleases = false)
-        super
-      end
-
-      def run(*args)
+      def run(@prereleases = false)
         return unless has_dependencies?
 
-        if lockfile?
-          manager.locks = locks
-          manager.resolve
-        else
-          manager.resolve
-        end
+        Shards.logger.info { "Resolving dependencies" }
 
-        manager.packages.each do |package|
-          analyze(package)
-        end
+        solver = Solver.new(spec, @prereleases)
+        solver.prepare(development: !Shards.production?)
 
-        if @up_to_date
-          Shards.logger.info "Dependencies are up to date!"
+        if packages = solver.solve
+          packages.each { |package| analyze(package) }
+
+          if @up_to_date
+            Shards.logger.info "Dependencies are up to date!"
```

Comment:
Why no block version as in other places? ditto below.

Reply:
It's a mere static string, it doesn't allocate anything.